src/java/boa/aggregators/Aggregator.java

```diff
@@ -122,6 +139,13 @@ protected void collect(final double data, final String metadata) throws IOExcept
 	protected void collect(final double data) throws IOException, InterruptedException {
 		this.collect(BoaCasts.doubleToString(data), null);
 	}
+	
+	public void aggregate(final Tuple data, final String metadata) throws IOException, InterruptedException, FinishedException, IllegalAccessException {	
```

Comment:
these should probably only be in the MLAggregator, since only those aggregators will utilize them

Reply:
I have used those aggregators to get the aggregator named by the emit key in "reduce" method in class BoaReducer. Should I move them to MLAggregator?

Reply:
I have moved all of them to MLAggregator.